### PR TITLE
Tweaks for islands off Vite

### PR DESCRIPTION
**Pull request by dmadisetti**
Just a working branch of trying to get islands off vite, corresponding marimo-quarto branch is here

I haven't had time to solidly work on this, but hope that I can address/ fix-up some of the issues tonight:

todos:
 - [x] isIsland
 - [x] clean-up

Bugs:
 - [ ] Islands with multiple outputs currently do not work
 - [x] json outputs still a little weird
 - [x] Dedent breaks functions
example: 
```python
def fn(x):
    pass
```
Could handle quarto side with easy hack of prepending "#\n", but asking render not to dedent is probably a good idea too

Beyond my control:
 - [ ] the island cdn is wishful thinking? https://cdn.jsdelivr.net/npm/@marimo-team/islands@0.4.3/dist/main.js

Things I would like to see:
 - [ ] intermediate flags (cache / pickle definitions for use in non-intermediate cells)
 - [ ] Quarto side, implement execution-options: https://quarto.org/docs/computations/execution-options.html

### CODE REVIEW

**Comment by dmadisetti** [comment]
Relevant quarto-marimo branch: https://github.com/dmadisetti/quarto-marimo/pull/3/files

**Comment by dmadisetti** [comment]
Just little changes to get things working from an external app.

There was 1 issue I wasn't able to track down, which is that multiple outputs seem broken during hydration (the worker sends down the right response)- but I'll open an issue for that

File: marimo/_islands/island_generator.py [hunk 1]
```diff
@@ -220,10 +239,24 @@ def render_head(
...
+            </marimo-version>
+            <marimo-mode data-mode="island" hidden=""></marimo-mode>
+
+            <!--
+              Required to silence marimo expectations, but not actively used
+              right now.
+            -->
+            <marimo-user-config data-config="{{}}" hidden="">
+            </marimo-user-config>
+            <marimo-app-config data-config="{{}}"> </marimo-app-config>
```
Comment (mscolnick) [inline marimo/_islands/island_generator.py:253]: i think its a bug if a codepath is using this / expecting this. if you remove this, does it still fail?

It should be handled by:
```
const {
  getMarimoVersion,
  getMarimoServerToken,
  getMarimoAppConfig,
  getMarimoUserConfig,
  getMarimoCode,
} = import.meta.env.VITE_MARIMO_ISLANDS
  ? islandsBasedMarimoSettings
  : domBasedMarimoSettings;
```

File: marimo/_islands/island_generator.py [hunk 2]
```diff
@@ -249,3 +282,17 @@ def render_head(
 
 def remove_empty_lines(text: str) -> str:
     return "\n".join([line for line in text.split("\n") if line.strip() != ""])
+
+
+def handle_mimetypes(output: CellOutput) -> str:
```
Comment (mscolnick) [inline marimo/_islands/island_generator.py:287]: you might be able to do `marimo.as_html(output.data).text`

File: marimo/_islands/island_generator.py (hunk 1 above)
Comment (dmadisetti) [reply marimo/_islands/island_generator.py:253]: Doesn't fail, but does produce error in console

Comment (mscolnick) [reply marimo/_islands/island_generator.py:253]: ah got it , thanks for explaining. PR looks good, thank you for the fixes!

**Review by mscolnick — APPROVED** [review]

File: marimo/_islands/island_generator.py (hunk 2 above)
Comment (dmadisetti) [reply marimo/_islands/island_generator.py:287]: Didn't handle images, and HTML was outputted literally- but I do this there's probably a better way

File: marimo/_islands/island_generator.py (hunk 1 above)
Comment (dmadisetti) [reply marimo/_islands/island_generator.py]: Woops, I missed there as well new commit should check for islands from mode.

So now normal marimo builds work for islands, separate islands builds aren't needed

Comment (mscolnick) [reply marimo/_islands/island_generator.py]: Ah nice - that's cool actually

**Comment by mscolnick** [comment]
@dmadisetti what do you mean by:

>  the island cdn is wishful thinking

If it was to publish it, I've done that task to this PR: https://github.com/marimo-team/marimo/pull/1230

**Comment by mscolnick** [comment]
and for:

>  Islands with multiple outputs currently do not work

Is this the global matplotlib issue or something else?

Btw this PR looks good to me, feel free to merge when you are done with it

**Review by mscolnick — APPROVED** [review]

**Comment by dmadisetti** [comment]
> @dmadisetti what do you mean by:
> 
> > the island cdn is wishful thinking
> 
> If it was to publish it, I've done that task to this PR: #1230

Wishful thinking in that the source currently points to a broken link. Cool! I'll try to publish the sample site once something is hosted

**Comment by dmadisetti** [comment]
For the multiple outputs bug see #1232

I don't have permissions to merge, but I think this is a good start!

**Comment by mscolnick** [comment]
thanks @dmadisetti, I will look into #1232 today

**Pull request merged**